### PR TITLE
[SPARK-14073][Streaming][test-maven]Move flume back to Spark

### PR DESCRIPTION
**Pull request by zsxwing**
## What changes were proposed in this pull request?

This PR moves flume back to Spark as per the discussion in the dev mail-list.
## How was this patch tested?

Existing Jenkins tests.

### CODE REVIEW

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

File: sql/core/src/test/java/test/org/apache/spark/sql/JavaDataFrameSuite.java [hunk 1]
```diff
@@ -303,27 +305,42 @@ public void pivot() {
...
   }
 
+  private String getResource(String resource) {
+    try {
+      // The following "getResource" has different behaviors in SBT and Maven.
+      // When running in Jenkins, the file path may contain "@" when there are multiple
+      // SparkPullRequestBuilders running in the same worker
+      // (e.g., /home/jenkins/workspace/SparkPullRequestBuilder@2)
+      // When running in SBT, "@" in the file path will be returned as "@", however,
+      // when running in Maven, "@" will be encoded as "%40".
```
Comment (zsxwing) [inline sql/core/src/test/java/test/org/apache/spark/sql/JavaDataFrameSuite.java]: I can reproduce this behavior by putting Spark to some folder containing "@" locally.

Comment (zsxwing) [inline sql/core/src/test/java/test/org/apache/spark/sql/JavaDataFrameSuite.java]: Here are the debug output I tried in Jenkins:

```
Running test.org.apache.spark.sql.JavaDataFrameSuite
raw:file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt
Find: /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/scala-2.11/test-classes/text-suite.txt
raw:file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt
Find: /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/scala-2.11/test-classes/text-suite.txt
raw:file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt
Find: /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/scala-2.11/test-classes/text-suite.txt
raw:file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt
Find: /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/scala-2.11/test-classes/text-suite.txt
Tests run: 17, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 7.628 sec <<< FAILURE! - in test.org.apache.spark.sql.JavaDataFrameSuite
testTextLoad(test.org.apache.spark.sql.JavaDataFrameSuite)  Time elapsed: 0.356 sec  <<< ERROR!
org.apache.spark.sql.AnalysisException: Path does not exist: file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt;
	at test.org.apache.spark.sql.JavaDataFrameSuite.testTextLoad(JavaDataFrameSuite.java:349)

testGenericLoad(test.org.apache.spark.sql.JavaDataFrameSuite)  Time elapsed: 0.322 sec  <<< ERROR!
org.apache.spark.sql.AnalysisException: Path does not exist: file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt;
	at test.org.apache.spark.sql.JavaDataFrameSuite.testGenericLoad(JavaDataFrameSuite.java:311)

Running test.org.apache.spark.sql.JavaRowSuite
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec - in test.org.apache.spark.sql.JavaRowSuite

Results :

Tests in error: 
  JavaDataFrameSuite.testGenericLoad:311 ? Analysis Path does not exist: file:/h...
  JavaDataFrameSuite.testTextLoad:349 ? Analysis Path does not exist: file:/home...
```
This is the line returned by `Thread.currentThread().getContextClassLoader().getResource("text-suite.txt").toString()`
``` 
file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt
```

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
> Test build #54054 has started for PR 11895 at commit f5d5976.

Hung by DataFrameReaderWriterSuite

retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by vanzin** [comment]
LGTM once we fix the maven test issues. Thanks for taking care of this!

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
Finally passed. Ping @rxin to take a look.

**Comment by rxin** [comment]
Thanks - merging in master.

**Pull request closed without merging**